index.md

```diff
@@ -374,6 +376,30 @@ describe('Array', function() {
 });
 ```
 
+## Retry Tests
+
+You can choose to retry a failed tests up to a certain number of times. This feature is designed to handle end-to-end tests (functional tests/Selenium...) where resources cannot be easily mocked/stubbed. **It's not recommended to use this feature for unit tests**.
+
+This feature does re-run `beforeEach/afterEach` hooks but not `before/after` hooks.
+
+```js
+describe('retries', function() {
+  // Retry all tests in this suite up to 4 times
+  this.retries(4);
+  
+  beforeEach(function () {
+    browser.get('http://www.yahoo.com');
+  });
+  
+  it('should succeed on the 3rd try', function () {
+    // This test specifically only retry up to 2 times
+    this.retries(2);
+    expect($('.foo').isDisplayed()).to.eventually.be.true;
```

Comment:
so for E2E frameworks like selenium webdriver, most accessors (or any operations) returns a `Promise`, in this case the `Promise` should eventually resolves to `true`. I picked this example since the `retry` feature is aimed at E2E tests :)